### PR TITLE
Dev aller

### PR DESCRIPTION
**Pull request by Aller05**
1. List sorting and search logic optimization:
   a. Sorting is supported within the folder;
   b. Supports superposition operations of tag filtering, keyword search, and sorting to ensure the correctness of list data display;
   c. The updateTime is not updated when the tag is modified; when the tag associated with the workflow is modified, the list is automatically refreshed;
2. The select all function supports flow within folders;
3. Add a new folder button next to the import button, click it to open the folder where the flow is saved locally;

### CODE REVIEW

File: ui/src/WorkspaceDB.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export function listFolderContent(
 }
 
 /** Class Workflow: below will be migrated to a class */
-export function updateFlow(id: string, input: Partial<Workflow>) {
+export function updateFlow(id: string, input: Partial<Workflow>, needUpdateTime = true) {
```
Comment (Weixuanf) [inline ui/src/WorkspaceDB.ts:135]: 我在想我们如果从input: Partial<Workflow>的field判断需不需要needUpdateTime呢？这样就不需要前端ui pass needUpdateTime了，直接在这里决定需不需要updateTime, 比如如果只是tag变了，就不用updatetime

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx [hunk 2]
```diff
@@ -75,69 +78,79 @@ export default function RecentFilesDrawer({ onclose, onClickNewFlow }: Props) {
...
+        if (selectedTag) {
+          tagFlag = !!flow.tags?.includes(selectedTag);
+        }
+        if (debounceSearchValue) {
+          keywordFlag = !!flow.name
+            .toLocaleLowerCase()
+            .includes(debounceSearchValue.toLocaleLowerCase());
+        }
+
+        return tagFlag && keywordFlag;
```
Comment (Weixuanf) [inline ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:105]: > b. Supports superposition operations of tag filtering, keyword search, and sorting to ensure the correctness of list data display;

i tried to search a tag name but it's not showing me any result

Comment (Weixuanf) [reply ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:105]: Screenshot 2024-01-07 at 5 13 55 PM
Screenshot 2024-01-07 at 5 14 03 PM

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx [hunk 3]
```diff
@@ -75,69 +78,79 @@ export default function RecentFilesDrawer({ onclose, onClickNewFlow }: Props) {
...
-  const onClickTag = (name: string) => {
-    setSelectedTag(name);
-    setFiles(listWorkflows().filter((n) => n.tags?.includes(name)));
-  };
-
   const loadLatestWorkflows = () => {
     const all = listFolderContent();
-    setFiles(sortFileItem(all, sortTypeRef.current));
+    aloneFlowsAndFoldersRef.current = all;
+    allFlowsRef.current = listWorkflows();
```
Comment (Weixuanf) [inline ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:84]: 没太看明白为什么我们需要这两个ref 是干啥的

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx [hunk 4]
```diff
@@ -75,69 +78,79 @@ export default function RecentFilesDrawer({ onclose, onClickNewFlow }: Props) {
...
+    (id: string) => {
+      deleteFlow(id);
+      loadLatestWorkflows();
+    },
+    [selectedTag, debounceSearchValue]
+  );
 
   useEffect(() => {
+    loadLatestWorkflows();
+    const handleDrop = (e: any) => {
```
Comment (Weixuanf) [inline ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:136]: good improvement! so handleDrop() won't be refreshed again and again 👍

File: ui/src/WorkspaceDB.ts (hunk 1 above)
Comment (Aller05) [reply ui/src/WorkspaceDB.ts:135]: 有道理，我按你说的改一下。

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx (hunk 2 above)
Comment (Aller05) [reply ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:105]: 搜索还是原来的名称模糊搜索，这里说的叠加操作，是原有的tag检索和你新增的搜索叠加操作，原来是不支持的。

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx (hunk 3 above)
Comment (Aller05) [reply ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:84]: 我看之前tag检索和名称模糊搜索，都是只检索flow，也只显示flow，不显示文件夹。

之前这两个检索，每次都是api获取全量flow，其实在没有新增或删除flow时，没必要每次都通过api获取。而且模糊搜索的调用频次也比较高。

所以在检索时，使用allFlowsRef来过滤后显示，取消检索时用aloneFlowsAndFoldersRef来还原列表。

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx (hunk 4 above)
Comment (Aller05) [reply ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:136]: 对滴。

File: ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx (hunk 3 above)
Comment (Weixuanf) [reply ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx:84]: 哦哦好的厉害了 我回头再看看

**Pull request merged**